### DISCUSSION TITLE
Question: which modem for Carbon X1 6gen?

### DISCUSSION THREAD

**Discussion by artemave** [danielewood/sierra-wireless-modems#15] [General] [answered]
Thanks for this amazing repo first of all.

From the readme it is not obvious which modem should be picked for Carbon X1 - hence the question.

**Comment by danielewood** [comment, marked as answer]
For newer generations of Lenovo laptops(Starting with Kaby Lake), they are doing something to make them only work with official Lenovo OEM Modems. **Note: This is according to forum discussions. I do not have a newer laptop to validate.**

For the X1 6th Generation, you need FRU 01AX792. But, make sure your specific SKU is WWAN-Ready and has both a SIM slot and antenna leads.

Someone is [selling this modem for $120 on eBay](https://www.ebay.com/itm/New-Lenovo-ThinkPad-Fibocom-L850-GL-WWAN-wireless-cellular-modem-4G-LTE-8TH-GEN/254224824463), seller says its compatible with the X1 Gen6, so eBay buyer protection has your back if it doesnt work.

Also see these discussions:
- https://forums.lenovo.com/t5/ThinkPad-X-Series-Laptops/X1-Carbon-6th-Gen-WWAN-SIM-not-detected/td-p/4128734
- https://forums.lenovo.com/t5/ThinkPad-T400-T500-and-newer-T/T470-WWAN-card/td-p/3631589

**Comment by artemave** [comment]
Thank you very much!

**Comment by artemave** [comment]
Update to my question: I am on Ubuntu and I have already tried a modem that looks very similar to the one you suggested - https://www.ebay.co.uk/itm/Lenovo-4XC0R38452-ThinkPad-Fibocom-L850-GL-CAT9-WWAN-Wireless-cellular/382951453060?epid=27022568916&hash=item5929af7184:g:MHAAAOSwk6hbaOnb

**Comment by danielewood** [comment]
> Update to my question: I am on Ubuntu and I have already tried a modem that looks very similar to the one you suggested - https://www.ebay.co.uk/itm/Lenovo-4XC0R38452-ThinkPad-Fibocom-L850-GL-CAT9-WWAN-Wireless-cellular/382951453060?epid=27022568916&hash=item5929af7184:g:MHAAAOSwk6hbaOnb

Ahh, Linux and Fibocom is a bad combo.

You are in luck. It looks like some others have figured out how to bypass the Lenovo BIOS checks altogether by making the Sierra Wireless modems not present themselves at all until they have fully booted (which will be about 15-20seconds after the BIOS is done).


```
# Enable Command Echos
ATE1

# Enable Advanced Commands
AT!ENTERCND="A710"

# Enable fast enumeration for warm boots only (Restarts)
AT!CUSTOM="FASTENUMEN",2

# Ignore the W_DISABLE signal pin
AT!PCOFFEN=2

# Save settings and reboot modem to apply
AT!RESET
```

References:
- [Reddit - Thinkpad X1C6/T480S: Sierra Wireless EM7455/EM7565 Tweaks](https://old.reddit.com/r/thinkpad/comments/bwmt20/thinkpad_x1c6t480s_sierra_wireless_em7455em7565/)
- [Reddit - Sierra Wireless EM7455 seems working with my Thinkpad X1C6](https://old.reddit.com/r/thinkpad/comments/a3yd2j/sierra_wireless_em7455_seems_working_with_my/)
- [Lenovo Forums - Getting Sierra EM7455 (and similar) to work on X1C6](https://forums.lenovo.com/t5/Ubuntu/Getting-Sierra-EM7455-and-similar-to-work-on-X1C6/td-p/4326043)

**Comment by artemave** [comment]
Yeah, I've read some of that and that's how I ended up here :)

To double check, is this the model I should buy? [Sierra Wireless EM7455 Dell Qualcomm 4G LTE Network WWAN NGFF Card Module GS](https://www.ebay.co.uk/itm/Sierra-Wireless-EM7455-Dell-Qualcomm-4G-LTE-Network-WWAN-NGFF-Card-Module-GS/283355990444?epid=4028493114&hash=item41f95549ac&enc=AQADAAADAFjVrDbVsZ8oH%2F8PNHtt9VX4%2Fw7FZcmMuqsX8uaFEduVgD6LxQAjcBaj3eZORqdT18%2FI2ueay5%2FCapK6IGiI6S2cwSn8SxQNZHkI5aFSE4ZKgZ8gCDQqshc6tDHa1I%2BYGbOIeYaKWtLQP1PlLhkxNtvWW2bwLkPN5BqHXciXXUkx832KRgGHgQz0VOi9qHPHUi6k2YFEsOaAltFICfLCcYTZq1i%2FzVo3qiC0KdbAuY5pT%2BUIOrsTKxa3Bdps13Ct4aqefFF%2B4CyaI7pFRv217VOyIJKGDFvrXLaR9JI1p3qXIDO8bFuUWWQxs%2Bo6gwJ5%2FzFySyJBPNtWCu1w3vxRxcyUUgIfIMS7SepWzAPTHemGopSWux6yC1%2FVTZNX7Vv0c00zGEFn7TStlw3BCvMFDBhXiPjrPomiJMCEaNSLATz5HT3MxNHV2sSicbfMfs5P0UZW42pkopZQTfngZb0J4TcfuyW5mQoFzZZRG7NezOfywAUg13S8aanuy9f%2F2UM5guKyYH%2By%2FjEke2d7W5fSieIxHwedMcNAp%2F%2BODsOm6BE1hVzWp2f2CN7Te9eZkpFZg1wlk2%2BW5EVQPzpLaKQeLckr0Ev1YkmGsGVu8HiQEX6%2FNpHuzOVavoYqnKm9V6DJOfYUtG%2B3s2Ke5LPF%2BXR%2B04ZNQQCNQ3xCTD0upv9h6iUc6jc9yWeblZhA1UA1%2FLXOTlXpwY66Dux3JfOgPAuHfPpVus%2BQKzKelkcAPyyeOCsOUhYpqK6n1gLKu3VpJ1mYSdP5r9Q7yXU5SG8QYPiQid4lhjwAaK07V264Nn92pzIQeKGnAvZ9bKSCflBnw66Xl7euG%2BFc9AlQU0w79qrAHfxRK%2B2398MetZw4eiIINnPXUM3xnFfSkh66NWYMOx8REV3JcQ7UOpfamhB%2FJHiv11Y%2F59cqnaTO2jsKJ5dRfShnikoVmVRQSkaqC0VkMZTV9RShpj9ZvzqdM1GGlM7Wq84kGKNutSWHY0hoknXFoz%2BStQDtyG8caKKaVWV3csCmsg%3D%3D&checksum=283355990444f6fb0f297aab4e319882d0550a069e1e)

I noticed it has a socket for gps antenna, something that's missing on the L850. Is that a problem?

Also, excuse my ignorance, but how do I apply those `AT` commands?

**Comment by danielewood** [comment]
Yes, that modem you linked should be fine.

As to applying the AT Commands, have you read my flashing guides? They explain everything or [link to other resources.](https://github.com/danielewood/sierra-wireless-modems/blob/master/README.md#create-accessible-com-port)


If you also buy one of these, it will make your flashing/setup procedure go much more smoothly as you will not have to do the sleep trick just to be able to run commands and such:
[WWAN/LTE Module NGFF M.2 to USB 3.0 Adapter Card with SIM Card Slot 6pin](https://www.ebay.co.uk/itm/WWAN-LTE-Module-NGFF-M-2-to-USB-3-0-Adapter-Card-with-SIM-Card-Slot-6pin/273671735178)

After you have updated and run all the commands, you can then move the modem back into your laptop.

I should probably test these settings and integrate them as default into my flashing script at some point.

**Comment by artemave** [comment]
Thank you! I'll read the flashing guided.
